src/boimmgpy/model_annotation/model_annotator.py

```diff
@@ -0,0 +1,431 @@
+        :rtype: Tuple(Dict)
+        """
+        results={}
+        counter = defaultdict(int)
+        get_synonym = self.get_synonym_id(self.backbone, side_chain)
+        if get_synonym != None and not get_synonym[2]:
+            backbone_id, sidechain_id, compound = get_synonym[0],get_synonym[1],get_synonym[2]
+            for backbone in backbone_id:
+                get_compound = self.get_compound(backbone, sidechain_id, compound)
+                if  get_compound != None and get_compound[0] != 0:
+                    backbone_coumpound, side_chains_compound = get_compound[0],get_compound[1]
+                    structurally_defined_lipids = (
+                        self.get_compounds_with_specific_parent_set_of_components(
+                         backbone_coumpound, side_chains_compound
+                        )
+                    )
+                    if structurally_defined_lipids:
+                        if metabolite.id in results:
+                            results[metabolite.id] = (
+                                results[metabolite.id]
+                                + structurally_defined_lipids
+                            )
+                        else:
+                            results[metabolite.id] = structurally_defined_lipids
+                        counter[self.backbone] += 1
+                if get_compound != None and get_compound[0] == 0:
+                    backbone_coumpound, side_chains_compound = get_compound[0],get_compound[1]
+                    results,counter = self.get_id_from_compound_with_no_backbone(backbone,side_chains_compound,metabolite,results,counter)
+
+        if get_synonym != None and get_synonym[2]:
+            backbone_id, sidechain_id, compound = get_synonym[0],get_synonym[1],get_synonym[2]
+            results,counter = self.get_id_from_compound_entitie(backbone_id,sidechain_id,compound,metabolite)
+        
+        
+        return counter,results
+    
+    def get_id_from_compound_with_no_backbone(self,backbone:str,side_chains_compound:List,metabolite:Metabolite,results:Dict,counter:Dict)->tuple():
+        """Method to handle compounds that have synonyms ID's for sidechains and backbone but only sidechains ID's for compound in database. This Method will implement a search directly for the 
+        generic compound with the same synonym as the backbone
+
+        :param backbone: backbone sinonym id
+        :type backbone: str
+        :param sidechain_id: sidechain synonym id
+        :type sidechain_id: list
+        :param metabolite: Metabolite being analized
+        :type metabolite: Metabolite
+        :return: tuple with two dictionaries,  first one for the extend of the lipid classes that the algorithm caugth and second one for the identifiers to do the annotation
+        :rtype: Tuple(Dict)
+        """
+        backbone_compound = self.get_compound_from_synonym(backbone)
```

Comment:
I got confused now because the **self.get_compound_from_synonym** method takes as input an integer and backbone is typed as a string. Check what is happening and correct